v21.1/computed-columns.md

```diff
@@ -28,7 +27,6 @@ Virtual computed columns:
 - Are not stored in the table's primary index.
 - Are recomputed as the column data in the expression changes.
 - Cannot be used as part of a primary key, `FAMILY` definition, in `CHECK` constraints, or `FOREIGN KEY` constraints.
-- Cannot be a [foreign key](foreign-key.html) reference.
```

Comment:
This restriction still remains. It's quite confusing and I had to double check myself. Here are examples of the differences:

A computed column expression can now reference a foreign key column, so this is allowed:

```sql
defaultdb> CREATE TABLE parent (p INT PRIMARY KEY);
CREATE TABLE

defaultdb> CREATE TABLE child (c INT PRIMARY KEY, c_p INT REFERENCES parent (p), v INT AS (c_p + 10) VIRTUAL);
CREATE TABLE
```

But this is still NOT allowed:

```sql
defaultdb> CREATE TABLE parent (p INT PRIMARY KEY, v INT AS (p + 10) VIRTUAL, UNIQUE INDEX (v))
CREATE TABLE

defaultdb> CREATE TABLE child (c INT PRIMARY KEY, v INT REFERENCES parent (v))
ERROR: unimplemented: virtual column "v" cannot be referenced by a foreign key
SQLSTATE: 0A000
HINT: You have attempted to use a feature that is not yet implemented.
See: https://go.crdb.dev/issue-v/59671/v22.1
```

Reply:
Oof, that's tricky. Took me a while to wrap my head around it, but I think I understand. Basically, you can create a computed column that references a foreign key, but you can't create a foreign key that references a computed column.

Done.